### ISSUE TITLE
Changing addParams in globalCommunicationParameter

### ISSUE THREAD

**Issue by mwinkelm** [FACT-Finder-Web-Components/ff-web-components#33]
Hi,

we recognized an issue with a current version of FF web components, in more detail with the `ff-communication` part. In our Angular setup we want to set a filter based on the given url. For example we have an url like https://example.com/something then we want to get the FF search results with a category filter without adding it to the url. Therefore we added it with `addParams` to `factfinder.communication.globalCommunicationParameter` where `factfinder` is a reference to the global FF object.

This works until version 3.11.3. In newer versions we always get all products (as the query is emtpy and defaults to *) and the filter is not applied, although we can see that the `addParams` field is filled correctly.

**Version 3.14.1**
Generated URL:
**https://...-ng.fact-finder.de/fact-finder/rest/v3/search/commerce_b2c_OCDKELG?query=*&sid=RRAujr0twv6XuokJFrwkcHXghztXfE&format=json**

GlobalCommunicationParameter:
![addParams-3-14-1](https://user-images.githubusercontent.com/6881829/80579330-226b8800-8a0a-11ea-83c8-fca0fa7dde8a.png)

**Version 3.11.3**
Generated URL:
**https://...-ng.fact-finder.de/fact-finder/rest/v3/search/commerce_b2c_OCDKELG?query=*&filter=ArticleAssignments.AssignmentSystem.Menu_Structure%3ATV+%26+Billede&sid=RRAujr0twv6XuokJFrwkcHXghztXfE&format=json**

GlobalCommunicationParameter:
![addParams-3-11-3](https://user-images.githubusercontent.com/6881829/80579340-27303c00-8a0a-11ea-8fed-2b1ae02e69b8.png)

Was that feature touched recently which might break anything concerning this topic?

Thanks,
Markus

**Comment by omikron-fp** [comment]
Hi @mwinkelm,
Sorry for the inconvenience. I don't think there were direct changes to the described feature but this might be a side effect of some internal changes that were made recently.
We will investigate and come back to you asap.

**Comment by KrzysztofKozlowskiOBSS** [comment]
Hi @mwinkelm,

`addParams` seems to be working fine in this very basic setup:
http://qa.web-components.fact-finder.de/web-components-test/demo/addParams/1504/test/elements/add-params.demo.html
 (please check the HTML source and the network tab).

Can you provide us with your code or test environment so we can investigate it further?
Also, what is the way you modify the `globalCommunicationParameter` object? Do you set only the specific properties individually like in the demo above, or do you override the entire object immutably?

**Comment by mwinkelm** [comment]
Our method (TypeScript) looks like this:
```javascript
public async setConfiguration(configuration: Partial<FactFinderGlobalCommunicationParameter>): Promise<void> {
    const factfinder = await this.factfinder;
    factfinder.communication.globalCommunicationParameter = {
      ...factfinder.communication.globalCommunicationParameter,
      ...configuration,
      useUrlParameter: false,
    };
  }
```
So, yes we are overrding the entire object.

This is called like this:
```javascript
await this.factfinder.setConfiguration({
        addParams: `${page}filter=ArticleAssignments.AssignmentSystem.Menu_Structure%3A${query}`,
      });
```
`${query}` is not the FACT-Finder query.

**Comment by KrzysztofKozlowskiOBSS** [comment]
@mwinkelm, due to some internal changes overriding the whole object no longer works. However, we find your code very valid and we are going to fix this on our side. In the meantime, you might use `Object.assign()` instead:

```javascript
public async setConfiguration(configuration: Partial<FactFinderGlobalCommunicationParameter>): Promise<void> {
    const factfinder = await this.factfinder;
    Object.assign(factfinder.communication.globalCommunicationParameter, {
      ...configuration,
      useUrlParameter: false,
    });
  }
```

**Comment by omikron-fp** [comment]
Hi @mwinkelm,
The fix is implemented and we are planning to release it later this week.

**Comment by omikron-fp** [comment]
Just released the fix as [3.15.0](https://github.com/FACT-Finder-Web-Components/ff-web-components/releases/tag/3.15.0)
